memory_test.go

```diff
@@ -287,164 +348,205 @@ func TestMemCheck(t *testing.T) {
 		},
 		{
-			"ArrayBuffer",
-			`var buffer = new ArrayBuffer(16);
+			desc: "ArrayBuffer",
+			script: `var buffer = new ArrayBuffer(16);
 
 			checkMem();
 			buffer2 = new ArrayBuffer(16);
 			checkMem();`,
-			7 + // "buffer2"
+			expectedSizeDiff: 7 + // "buffer2"
 				16 + // data size
-				SizeEmpty + SizeEmpty, // base object + prototype
+				SizeEmptyStruct + SizeEmptyStruct, // base object + prototype
+			expectedNewSizeDiff: (7 + SizeString) + // "buffer2"
+				16 + // data size
+				SizeEmptyStruct + SizeEmptyStruct, // base object + prototype,
 		},
 		{
-			"DataView",
-			`var buffer = new ArrayBuffer(16);
+			desc: "DataView",
+			script: `var buffer = new ArrayBuffer(16);
 			var view = new DataView(buffer, 0);
 			var buffer2 = new ArrayBuffer(16);
 
 			checkMem();
 			view2 = new DataView(buffer2, 0);
 			checkMem();`,
-			5 + // "view2"
-				SizeEmpty + // DataView overhead
-				SizeEmpty + SizeEmpty + // base object + prototype
-				SizeEmpty, // array buffer reference
-		},
-		{
-			"Number",
-			`num1 = new Number("1")
+			expectedSizeDiff: 5 + // "view2"
+				SizeEmptyStruct + // DataView overhead
+				SizeEmptyStruct + SizeEmptyStruct + // base object + prototype
+				SizeEmptyStruct, // array buffer reference
+			expectedNewSizeDiff: (5 + SizeString) + // "view2"
+				SizeEmptyStruct + // DataView overhead
+				SizeEmptyStruct + SizeEmptyStruct + // base object + prototype
+				SizeEmptyStruct, // array buffer reference,
+		},
+		{
+			desc: "Number",
+			script: `num1 = new Number("1")
 
 			checkMem();
 			num2 = new Number("2")
 			checkMem();`,
-			4 + // "num2"
+			expectedSizeDiff: 4 + // "num2"
+				SizeNumber +
+				SizeEmptyStruct + SizeEmptyStruct, // base object + prototype
+			expectedNewSizeDiff: (4 + SizeString) + // "num2"
 				SizeNumber +
-				SizeEmpty + SizeEmpty, // base object + prototype
+				SizeEmptyStruct + SizeEmptyStruct, // base object + prototype,
 		},
 		// TODO(REALMC-10739) add a test that calls Error.captureStackTrace when it is implemented)
 		{
-			"stash",
-			`checkMem();
+			desc: "stash",
+			script: `checkMem();
 			try {
 				throw new Error("abc");
 			} catch(e) {
 				checkMem();
 			}
 			`,
-			7 + 3 + // Error "message" field + len("abc")
+			expectedSizeDiff: 7 + 3 + // Error "message" field + len("abc")
 				4 + 5 + // Error "name" field + len("Error")
-				SizeEmpty + SizeEmpty, // base object + prototype
+				SizeEmptyStruct + SizeEmptyStruct, // base object + prototype
+			expectedNewSizeDiff: (7 + SizeString) + (3 + SizeString) + // Error "message" field + len("abc")
+				(4 + SizeString) + (5 + SizeString) + // Error "name" field + len("Error")
+				SizeEmptyStruct + SizeEmptyStruct, // base object + prototype,
 		},
 		{
-			"Native value",
-			`checkMem();
+			desc: "Native value",
+			script: `checkMem();
 			nv = new MyNativeVal()
 			checkMem();`,
-			testNativeValueMemUsage + 2,
+			expectedSizeDiff: testNativeValueMemUsage +
+				2, // "nv"
```

Comment:
[q] what does `nv` mean here? Native Value?

Reply:
this is for the variable name `nv`, so it's the len("nv")